tree_test.go

```diff
@@ -351,38 +352,44 @@ func TestFlush1kLeaves(t *testing.T) {
 	}
 }
 
-func BenchmarkCommit1kLeaves(b *testing.B) {
-	benchmarkCommitNLeaves(b, 1000)
-}
-
-func BenchmarkCommit10kLeaves(b *testing.B) {
-	benchmarkCommitNLeaves(b, 10000)
+func BenchmarkCommitLeaves(b *testing.B) {
+	benchmarkCommitNLeaves(b, 1000, 10)
+	benchmarkCommitNLeaves(b, 10000, 10)
+	benchmarkCommitNLeaves(b, 1000, 8)
+	benchmarkCommitNLeaves(b, 10000, 8)
 }
 
 func BenchmarkCommitFullNode(b *testing.B) {
-	value := []byte("value")
-	keys := make([][]byte, 1024)
-	for i := 0; i < 1024; i++ {
-		key := make([]byte, 32)
-		binary.BigEndian.PutUint16(key[:2], uint16(i)<<6)
-		keys[i] = key
-	}
+	benchmarkCommitFullNode(b, 10)
+	benchmarkCommitFullNode(b, 8)
+}
+func benchmarkCommitFullNode(b *testing.B, width int) {
+	b.Run(fmt.Sprintf("width/%d", width), func(b *testing.B) {
+		nChildren := 1 << width
+		value := []byte("value")
+		keys := make([][]byte, nChildren)
+		for i := 0; i < nChildren; i++ {
+			key := make([]byte, 32)
+			binary.BigEndian.PutUint16(key[:2], uint16(i)<<6)
```

Comment:
This only works for width 10. For width 8 I imagine it has to be something like:

```go
key[0] = byte(i)
```